From e25cbd5234bfdbf771ece687aa5cf664a980175d Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Tue, 6 Jan 2009 10:06:17 +0800
Subject: [PATCH] ENGR00103955 Add control for WIFI and MMC 8bit

Use the MUX4_CTL of the MCU to control the SDHC1 8bit
or the SD2 WIFI

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/mach-mx35/mx35_3stack_gpio.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx35/mx35_3stack_gpio.c b/arch/arm/mach-mx35/mx35_3stack_gpio.c
index d2d4608..ef5630e 100644
--- a/arch/arm/mach-mx35/mx35_3stack_gpio.c
+++ b/arch/arm/mach-mx35/mx35_3stack_gpio.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -545,6 +545,10 @@ void gpio_sdhc_active(int module)
 				  MUX_CONFIG_FUNC | MUX_CONFIG_SION);
 #if defined(CONFIG_SDIO_UNIFI_FS) || defined(CONFIG_SDIO_UNIFI_FS_MODULE)
 #else
+		/* MUX4_CTR , 0: SD2 to WIFI, 1:SD2 to SD1 8bit */
+		if (board_is_mx35(BOARD_REV_2))
+			pmic_gpio_set_bit_val(MCU_GPIO_REG_GPIO_CONTROL_2,
+					      7, 1);
 		mxc_request_iomux(MX35_PIN_SD2_CMD,
 				  MUX_CONFIG_ALT2 | MUX_CONFIG_SION);
 		mxc_request_iomux(MX35_PIN_SD2_CLK,
@@ -583,6 +587,10 @@ void gpio_sdhc_active(int module)
 #endif
 		break;
 	case 1:
+		/* MUX4_CTR , 0: SD2 to WIFI, 1:SD2 to SD1 8bit */
+		if (board_is_mx35(BOARD_REV_2))
+			pmic_gpio_set_bit_val(MCU_GPIO_REG_GPIO_CONTROL_2,
+					      7, 0);
 		mxc_request_iomux(MX35_PIN_SD2_CLK,
 				  MUX_CONFIG_FUNC | MUX_CONFIG_SION);
 		mxc_request_iomux(MX35_PIN_SD2_CMD,
-- 
1.5.4.4

